packages/email-notification-dispatcher/src/services/emailNotificationDispatcherService.ts

```diff
@@ -0,0 +1,29 @@
+import { EmailNotificationMessagePayload } from "pagopa-interop-models";
+import { initProducer } from "kafka-iam-auth";
+import { config } from "../config/config.js";
+import { encodeEmailEvent } from "../handlers/utils.js";
+
+const producer = await initProducer(config, config.emailSenderTopic);
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function emailNotificationDispatcherServiceBuilder() {
+  return {
+    sendMessage: async (
+      streamId: string,
+      messagePayload: EmailNotificationMessagePayload
+    ): Promise<void> => {
+      await producer.send({
+        messages: [
+          {
+            key: streamId,
```

Comment:
I'm not sure if it's useful to set a `key` here (I think it's used for partitioning and not required, reading the docs: https://kafka.js.org/docs/producing#message-key) and if so what to use

using stream id would mean that messages for the same event or different events for the same entity (e.g. the same agreement) are placed in the same partition and sent in order

but anyway we can't guarantee they'll be delivered in order, so maybe we could just not set the key at all